tests/webrtc.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 litep2p developers
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use futures::StreamExt;
+use litep2p::{
+    config::ConfigBuilder as Litep2pConfigBuilder,
+    crypto::ed25519::Keypair,
+    protocol::{libp2p::ping, notification::ConfigBuilder},
+    transport::webrtc::config::Config,
+    types::protocol::ProtocolName,
+    Litep2p,
+};
+
+#[tokio::test]
+#[ignore]
+async fn webrtc_test() {
+    let _ = tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .try_init();
+
+    let (ping_config, mut ping_event_stream) = ping::Config::default();
+    let (notif_config, mut notif_event_stream) = ConfigBuilder::new(ProtocolName::from(
+        // "/c0c89622f83f6f3c6b94bc307fec1652a3aa58ac88a564c34706633f44cbb3d1/block-announces/1",
+        "/e143f23803ac50e8f6f8e62695d1ce9e4e1d68aa36c1cd2cfd15340213f3423e/block-announces/1",
+    ))
+    .with_max_size(5 * 1024 * 1024)
+    .with_handshake(vec![1, 2, 3, 4])
+    .with_auto_accept_inbound(true)
+    .build();
+
+    let config = Litep2pConfigBuilder::new()
+        .with_keypair(Keypair::generate())
+        .with_webrtc(Config {
+            listen_addresses: vec!["/ip4/192.168.1.170/udp/8888/webrtc-direct".parse().unwrap()],
```

Comment:
The block-announces protocol uses `e143f23803ac50e8f6f8e62695d1ce9e4e1d68aa36c1cd2cfd15340213f3423e` which is the genesis of Westend, however I cannot connect to this addr.
Is there another way to test this?

Reply:
The listen address should be updated to 127.0.0.1. Or does the WebRTC spec impose the limitations of connecting to loopback addresses?

Then, I guess, for this test to succeed another Westend node with litep2p backend is needed configured to dial this listen address.

Reply:
Taking into account the noise prologue comment, it looks like smoldot is needed as a second node for this test to succeed.

Reply:
Thanks @altonen for the explanation!